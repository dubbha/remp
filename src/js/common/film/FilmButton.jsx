import React from 'react';
import PropTypes from 'prop-types';

const SearchButton = ({ onSearchClick }) => (
  <button
    className="film__button"
    onClick={onSearchClick}
  >
    Search
  </button>
);

SearchButton.propTypes = {
  onSearchClick: PropTypes.func.isRequired,
};

export default SearchButton;
