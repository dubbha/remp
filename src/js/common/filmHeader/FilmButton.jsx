import React from 'react';
import PropTypes from 'prop-types';

const FilmButton = ({ onSearchClick }) => (
  <button
    className="film__button"
    onClick={onSearchClick}
  >
    Search
  </button>
);

FilmButton.propTypes = {
  onSearchClick: PropTypes.func.isRequired,
};

export default FilmButton;
