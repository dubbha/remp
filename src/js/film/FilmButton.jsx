import React from 'react';
import PropTypes from 'prop-types';

const SearchButton = ({ onClearFilm }) => (
  <button
    className="film__button"
    onClick={onClearFilm}
  >
    Search
  </button>
);

SearchButton.propTypes = {
  onClearFilm: PropTypes.func.isRequired,
};

export default SearchButton;
