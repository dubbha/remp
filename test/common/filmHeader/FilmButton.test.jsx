import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FilmButton from 'common/filmHeader/FilmButton';

describe('FilmButton', () => {
  it('should render successfully', () => {
    const props = { onSearchClick: jest.fn() };
    const wrapper = shallow(<FilmButton {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
