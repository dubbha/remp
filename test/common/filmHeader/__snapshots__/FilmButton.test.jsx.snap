// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmButton should render successfully 1`] = `
<button
  className="film__button"
  onClick={[Function]}
>
  Search
</button>
`;
